pizzeria-frontend/src/pages/basketPage.tsx

```diff
@@ -1,35 +1,207 @@
-import React, { useEffect } from "react";
-import { ProductProps } from "../types";
+import React, { useEffect, useRef, useState } from "react";
+import { BasketProps } from "../types";
 import Cookies from "js-cookie";
 import NavBar from "../NavBar";
 import { useBasketContent } from "../providers/BasketContentProvider";
+import pizza from "../images/pizza.jpg";
+import { useBasket } from "../providers/BasketCounterProvider";
+
+function round(num: number, fractionDigits: number): number {
+  return Number(num.toFixed(fractionDigits));
+}
 
 const BasketPage = () => {
   const { basketItems, setBasketItems } = useBasketContent();
 
-  useEffect(() => {
-    setBasketItems(JSON.parse(Cookies.get("basket") || "[]"));
-  }, [basketItems]);
   return (
     <div>
       <NavBar></NavBar>
-      <div className="flex flex-row">
+      <div className="flex flex-row justify-around">
+        <div className="flex flex-col">
+          {basketItems.length ? (
+            basketItems.map((item) => {
+              return (
+                <ProductOrdered
+                  item={item}
+                  key={item.product.id}
+                ></ProductOrdered>
+              );
+            })
+          ) : (
+            <div>NO items in basket</div>
+          )}
+
+          <Checkout></Checkout>
+        </div>
         <div className="flex flex-col">
-          {basketItems.map((item) => {
-            return (
-              <ProductOrdered
-                product={item.product}
-                key={item.product.id}
-              ></ProductOrdered>
-            );
-          })}
+          <label>More information about the order:</label>
+          <textarea
+            className="border border-slate-300"
+            placeholder="IDK"
+          ></textarea>
         </div>
       </div>
     </div>
   );
 };
-const ProductOrdered = (props: ProductProps) => {
-  return <div className=""></div>;
+
+const ProductOrdered = (props: BasketProps) => {
+  const { basketItems, setBasketItems } = useBasketContent();
+  const { setBasketCounter } = useBasket();
+  const IsButtonDisabled = props.item.amount === 1 ? true:false
+
+  useEffect(() => {
+    console.log("basketItems:");
+
+    setBasketCounter(basketItems.length);
+
+    const basketString = JSON.stringify(basketItems);
+
+    Cookies.set("basket", basketString);
+
+  }, [basketItems, setBasketCounter]);
```

Comment:
Tested it now, it does not console log endlessly if that means something. I think it could have happened if I changed the basketItems array in this useEffect though

Reply:
Yup, if it doesnt consoleLog endlessly its good